### PR TITLE
[#171608911] Set pin lenght to six digits

### PR DESCRIPTION
**Pull request by Edigio**
2 Features:
- Set the pin length from 5 to 6 digits
- Render the numbers present on the pinpad randomly

### CODE REVIEW

**Comment by Edigio** [comment]
This is the result

 <table style="width:100%">
  <tr>
    <th>iPhone X</th>
    <th>iPhone 5s</th>
  </tr>
<tr>
    <td width="320"><image src="https://user-images.githubusercontent.com/7993551/76771548-cde4b400-679f-11ea-9486-db143da53927.png"/></td>
    <td width="320"><image src="https://user-images.githubusercontent.com/7993551/76771516-c2918880-679f-11ea-867a-1e3de43edb38.png"/></td>
  </tr>
</table>

File: ts/components/Pinpad/index.tsx [hunk 1]
```diff
@@ -144,13 +157,56 @@ class Pinpad extends React.PureComponent<Props, State> {
...
+      return Math.random() - 0.5;
+    });
+
+    if (
+      (this.props.compareWithCode &&
+        this.props.compareWithCode.length !== PIN_LENGTH) ||
+      this.props.compareWithCode === undefined
+    ) {
+      // tslint:disable-next-line: no-object-mutation
+      this.placeholderPositions = [...new Array(NEW_PIN_LENGTH)];
```
Comment (Undermaken) [inline ts/components/Pinpad/index.tsx]: You can remove this.
In render method, where it is used, you can replace with this

```typescript
{range(0, this.state.pinLength).map(this.renderPlaceholder)}
```

File: ts/components/Pinpad/index.tsx [hunk 2]
```diff
@@ -88,41 +97,45 @@ class Pinpad extends React.PureComponent<Props, State> {
...
-    [
-      this.props.onCancel
-        ? Tuple2(
-            I18n.t("global.buttons.cancel").toUpperCase(),
-            this.props.onCancel
-          )
-        : this.props.isFingerprintEnabled &&
-          this.props.biometryType &&
-          this.props.onFingerPrintReq
+  private pinPadDigits = (): ComponentProps<typeof KeyPad>["digits"] => {
```
Comment (Undermaken) [inline ts/components/Pinpad/index.tsx]: With this logic **0 digit** holds ever the same position

File: ts/components/Pinpad/index.tsx [hunk 3]
```diff
@@ -144,13 +157,56 @@ class Pinpad extends React.PureComponent<Props, State> {
...
   constructor(props: Props) {
     super(props);
-
+    this.placeholderPositions = [...new Array(PIN_LENGTH)];
     this.state = {
       value: "",
-      isDisabled: false
+      isDisabled: false,
+      pinLength: PIN_LENGTH,
+      pinPadValue: new Array("1", "2", "3", "4", "5", "6", "7", "8", "9"),
```
[the remark this answers is not in the document]
Comment (Edigio) [reply ts/components/Pinpad/index.tsx]: `range(1, 9)` otherwise it also inserts 10

**Pull request merged**